### PR TITLE
add docker image build + push to CI

### PR DESCRIPTION
**Pull request by gbrik**

### CODE REVIEW

**Comment by gbrik** [comment]
couple questions:
- what are the preferred credentials for pushing to docker?
- is there a way to dryrun a .travis.yml? i tested locally that goreleaser will build the images correctly, but don't know that it all fits together

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,7 @@
+FROM sourcegraph/alpine:3.10
+
+# needed for `src lsif upload`
```
Comment (eseliger) [inline Dockerfile:3]:
```suggestion
# needed for `src lsif upload` and `src actions exec`
```

**Comment by slimsag** [comment]
> is there a way to dryrun a .travis.yml? i tested locally that goreleaser will build the images correctly, but don't know that it all fits together

not easily, theoretically you could configure .travis.yml to run on non-master branches but its more trouble than its worth, and in cases like this I recommend just doing a best-effort then testing in `master`.

Our buildkite CI in our main repository has better dry-run capabilities, so things are better/different there.

File: .travis.yml [hunk 2]
```diff
@@ -13,7 +15,9 @@ env:
   - secure: eFuG5PSCXNz0i4Knic0wVmTko+O/epoLWpDtDs7jGHo3UD0JIukDTeIoHRPF56LEF3ZmfJWarc0XWZRSUmtn2EYydvCrTlroXLe6lfiFL/ctH0CWpArCp5xHQKVqKk+h6F53hAva3o181PfnFtk58H/xgj2jo8sANbqXricnL5aT34RVje85483poL2RLmd2WiYVqDqeGdkg7DaorpJra1mFvOCmInrbMpexpomJKmpCbxzrdvUmvRE9bDWLHctgvWL6bVTsQNzUN0zh6AIfT0FuFmiwjVlD5d4Bo10jUucg1qs4gglHneplPMoQb3dW7reeE1hIjuZSjXmDZKwYaonpuy8lsEJTEZpo18obYfRP9fjJZJb71rQ/sjbAd2EEV84r13k4JcprUp4uFd/LgUpHNwyRxnm2bKneHSpf9eb0D+A6Kzxkix/vSJeH8G/pHc07mYtf1rqMG57eVeeVinAFbGsvmggQfFGXupW3c48VYS6TTUmcUv0B7jiDHEdFG1zRXElX48S6HJnhI4531jxbEsvgBt63sR5wH5dZv++VodzznwjaJ4LJv+bxZEtswhNA/nzxGCk9vNDkhPxpS531mNAyfV2ewJ81QW94BuWh0dTfo6gVypCS/2xI8L2I2sqs5vk2iCr51WJDL79yLXRP6QhXLcr8jgDpBQMNbpE=
 deploy:
 - provider: script
-  script: curl -sL https://git.io/goreleaser | bash
+  script:
+    - echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
```
Comment (slimsag) [inline .travis.yml:19]: Note: You'll need to add this to the env/global secure section above as a Travis encrypted secret. Username can be hard-coded here in plaintext though.

**Pull request merged**